docs/python/notebooks/onnxruntime-nuphar-tutorial.ipynb

```diff
@@ -0,0 +1,681 @@
+    "convert_to_scan_model(lstm_model, scan_model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After conversion, let's compare performance and accuracy with baseline:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'nuphar: 2.5135028000000013 seconds, baseline: 2.5710184 seconds'"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "sess_nuphar = onnxruntime.InferenceSession(scan_model)\n",
+    "output_nuphar = sess_nuphar.run([], feed)\n",
+    "assert np.allclose(output[0], output_nuphar[0])\n",
+    "\n",
+    "repeats = 10\n",
+    "start_baseline = timer()\n",
+    "for i in range(repeats):\n",
+    "    output = sess_baseline.run([], feed)\n",
+    "end_baseline = timer()\n",
+    "\n",
+    "start_nuphar = timer()\n",
+    "for i in range(repeats):\n",
+    "    output = sess_nuphar.run([], feed)\n",
+    "end_nuphar = timer()\n",
+    "\n",
+    "'nuphar: {} seconds, baseline: {} seconds'.format(end_nuphar - start_nuphar, end_baseline - start_baseline)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Quantize LSTM model\n",
+    "Let's get more speed-ups from Nuphar by quantizing the floating point GEMM/GEMV in LSTM model to int8 GEMM/GEMV.\n",
+    "\n",
+    "**NOTE:** For quantization to be fast, a CPU with AVX2 instructions is preferred."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'No AVX2, quantization model might be slow'"
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "subprocess.run([sys.executable, '-m', 'pip', 'install', '--user', 'cpufeature'])\n",
+    "import cpufeature\n",
+    "cpufeature.CPUFeature['AVX2'] or 'No AVX2, quantization model might be slow'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can use onnxruntime.nuphar.model_quantizer to quantize floating point GEMM/GEMVs. Assuming GEMM/GEMV takes form of input * weights, weights are statically quantized per-column, and inputs are dynamically quantized per-row."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "from onnxruntime.nuphar.model_quantizer import convert_matmul_model\n",
+    "quantized_model = 'Scan_LSTMx4_int8.onnx'\n",
+    "convert_matmul_model(scan_model, quantized_model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now run the quantized model, and check accuracy. Please note that quantization brings some error, so we relaxed on compare threshold."
```

Comment:
"quantization brings some error, so..." -> "quantization may cause losses of accuracy, so we relax the comparison threshold a bit."?